src/new_index/schema.rs

```diff
@@ -273,13 +280,53 @@ impl Indexer {
         let tip = daemon.getbestblockhash()?;
         let new_headers = self.get_new_headers(&daemon, &tip)?;
 
+        // Must rollback blocks before rolling forward
+        let headers_len = {
+            let mut headers = self.store.indexed_headers.write().unwrap();
+            let reorged = headers.apply(new_headers.clone());
+            assert_eq!(tip, *headers.tip());
+            let headers_len = headers.len();
+            drop(headers);
+
+            if !reorged.is_empty() {
+                if reorged.len() > 10 {
+                    warn!("reorg of over 10 blocks detected! Wonky stuff might happen!");
+                }
+                let (tx, rx) = crossbeam_channel::unbounded();
+                // Delete txstore
+                start_fetcher(self.from, &daemon, reorged.clone())?
+                    .map(|blocks| self.add(&blocks, Operation::DeleteBlocks));
+                // Delete history_db
+                start_fetcher(self.from, &daemon, reorged)?.map(|blocks| {
+                    self.index(&blocks, Operation::DeleteBlocksWithHistory(tx.clone()))
+                });
```

Comment:
is it possible for a reorg to be detected while we're still doing the initial sync (i.e. while `self.from == FetchFrom::BlkFiles`)?

if so, does this mean the `blkfiles_fetcher` would have to go back and re-parse every block file twice here to fetch the reorged ones? for the two delete operations?

I suppose it's a fairly unlikely edge case, but I'm just imagining the initial sync getting stuck reading all of the testnet3 block files forever because the chain keeps getting reorged every 10 minutes 😅

Reply:
oh, I suppose that could only happen once at most each time electrs gets restarted, since immediately after this section we run the indexing all the way through the finish and then set the "initial sync completed" flag in the db 🤔

Reply:
The headers get fetched once and then the initial block files sync runs for a couple days.

The headers are synced from Bitcoind before the first update call anyways, so pulling this check to the front might happen if a reorg happens just between when the headers sync in the init function of the headerslist and when its called inside update...

Not sure how to test that edge case... but it should be rare enough to not matter imo.

Reply:
yeah I think it would only happen in practice if you rebooted electrs partway through the initial sync, which should never really need to happen in production anyway.